### PR TITLE
Add build configuration for VESNIN server

### PR DESCRIPTION
**Pull request by artemsen**
Signed-off-by: Artem Senichev <a.senichev@yadro.com>

### CODE REVIEW

**Comment by stewart-ibm** [comment]
Can one of the admins verify this patch for testing?

**Comment by stewart-ibm** [comment]
ok to test

**Comment by stewart-ibm** [comment]
@wghoffa @dsanner do either of you have any comment/thoughts on the machine xml?

**Comment by stewart-ibm** [comment]
@artemsen This looks pretty good to me.
I've gone and created/forked https://github.com/open-power/vesnin-xml from your repo and sent you an invite on it.

Assuming Bill/Dean don't find anything obviously wrong with the XML, I'm happy to merge this (pointing at the open-power/vesnin-xml repo).

File: openpower/configs/vesnin_defconfig [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+BR2_LINUX_KERNEL_CUSTOM_VERSION_VALUE="4.15.6"
+BR2_LINUX_KERNEL_PATCH="$(BR2_EXTERNAL_OP_BUILD_PATH)/linux"
+BR2_LINUX_KERNEL_USE_CUSTOM_CONFIG=y
+BR2_LINUX_KERNEL_CUSTOM_CONFIG_FILE="$(BR2_EXTERNAL_OP_BUILD_PATH)/configs/linux/skiroot_defconfig"
+BR2_LINUX_KERNEL_ZIMAGE_EPAPR=y
+BR2_PACKAGE_BUSYBOX_CONFIG_FRAGMENT_FILES="$(BR2_EXTERNAL_OP_BUILD_PATH)/configs/busybox.fragment"
+BR2_PACKAGE_BUSYBOX_SHOW_OTHERS=y
+BR2_PACKAGE_LINUX_FIRMWARE=y
+BR2_PACKAGE_LINUX_FIRMWARE_BNX2X=y
+BR2_PACKAGE_LINUX_FIRMWARE_CXGB4_T4=y
```
Comment (shenki) [inline openpower/configs/vesnin_defconfig]: Does your system contain a Chelsio 10Gb Ethernet card and a Broadcom NetExtreme II? If not, you can remove these two lines and shrink the size of the skiroot environment a bit.

File: openpower/configs/hostboot/vesnin.config [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+set   ALLOW_MICRON_PNOR
+set   ALLOW_MACRONIX_PNOR
+
+# VPD options.
+set MVPD_READ_FROM_HW
+set MVPD_WRITE_TO_HW
+set MVPD_READ_FROM_PNOR
+set MVPD_WRITE_TO_PNOR
+set DJVPD_READ_FROM_HW
+set DJVPD_WRITE_TO_HW
```
Comment (dcrowell77) [inline openpower/configs/hostboot/vesnin.config]: FYI on the DJVPD_WRITE_TO_HW flag.  We have heard that some of our partners don't like this set.  Basically this allows us to store lane repair information for bad memory bits.  If we write into the DIMM then the repair will follow the DIMM around.  We like that for our systems but some other folks don't, especially in early bringup of the system where you might hit false errors due to board issues.

File: openpower/configs/hostboot/vesnin.config [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+set PVPD_READ_FROM_HW
+set PVPD_WRITE_TO_HW
+set PVPD_READ_FROM_PNOR
+set PVPD_WRITE_TO_PNOR
+set SKIP_RESTRICT_EX_UNITS
+unset CDIMM_FORMAT_FOR_CVPD
+
+# gpio config
+set GPIODD
+set PALMETTO_VDDR
```
Comment (dcrowell77) [inline openpower/configs/hostboot/vesnin.config]: I don't think you want PALMETTO_VDDR set, but this is really deprecated anyway so it doesn't do anything

File: openpower/configs/hostboot/vesnin.config [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+# Enable SBE updates
+set SBE_UPDATE_INDEPENDENT
```
Comment (dcrowell77) [inline openpower/configs/hostboot/vesnin.config]: Our recommendation for SBE updates in P9 is to use SBE_UPDATE_CONSECUTIVE.  We don't support the previous concept of a 'golden' seeprom.

File: openpower/configs/hostboot/vesnin.config [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+unset PCIE_HOTPLUG_CONTROLLER
+
+# turn on console output
+set CONSOLE
+set BMC_AST2400
+
+# Enable Kingston dimm voltage workaround
+set KINGSTON_1_35_VOLT
```
Comment (dcrowell77) [inline openpower/configs/hostboot/vesnin.config]: fyi - this value is not used anymore

**Comment by dcrowell77** [comment]
Just curious - how did you generate your vesnin.xml file?

**Comment by artemsen** [comment]
@dcrowell77 Originally all these files were created by MSI.

File: openpower/configs/hostboot/vesnin.config (hunk 2 above)
Comment (artemsen) [reply openpower/configs/hostboot/vesnin.config]: Thanks, we will keep it in mind.

**Comment by stewart-ibm** [comment]
Thanks, merged!

I switched the defconfig to point to the open-power/vesnin-xml repository when I merged it.

**Pull request merged**